lib/protip/wrapper.rb

```diff
@@ -235,44 +246,36 @@ def matches?(field, value)
 
     end
 
-    def method_missing_oneof(oneof_descriptor)
-      oneof_field_name = message.send(oneof_descriptor.name)
-      return if oneof_field_name.nil?
-      oneof_field_name = oneof_field_name.to_s
-      oneof_field = oneof_descriptor.detect {|field| field.name == oneof_field_name}
-      oneof_field ? get(oneof_field) : nil
+    def method_missing_oneof(oneof, *args)
+      raise ArgumentError unless args.length == 0
+      field_name = message.public_send(oneof.name)
+
+      field_name ? get(message.class.descriptor.lookup(field_name.to_s)) : nil
     end
 
     def method_missing_field(field, *args)
-      if field
-        raise ArgumentError unless args.length == 0
-        get(field)
-      end
+      raise ArgumentError unless args.length == 0
+      get field
     end
 
     def method_missing_query(name, *args)
-      field = message.class.descriptor.detect do |field|
-        self.class.matchable?(field) && :"#{field.name}?" == name
-      end
-      if args.length == 1
-        # this is an enum query, e.g. `state?(:CREATED)`
-        matches? field, args[0]
-      elsif args.length == 0
-        # this is a boolean query, e.g. `approved?`
-        get field
+      field = message.class.descriptor.lookup(name[0, name.length - 1])
+      raise NoMethodError if !field || !self.class.matchable?(field)
+      if field.type == :enum
+        raise ArgumentError unless args.length == 1
+        return matches?(field, args[0])
+      elsif field.type == :bool ||
+        (field.type == :message && field.subtype.name == 'google.protobuf.BoolValue')
       else
-        raise ArgumentError
+        raise NoMethodError
       end
     end
 
-    def method_missing_setter(name, *args)
-      field = message.class.descriptor.detect{|field| :"#{field.name}=" == name}
-      if field
-        raise ArgumentError unless args.length == 1
-        attributes = {}.tap { |hash| hash[field.name] = args[0] }
-        assign_attributes attributes
-        return args[0] # return the input value (to match ActiveRecord behavior)
-      end
+    def method_missing_set(name, *args)
```

Comment:
Also @tubergen do you remember why you were doing this `.tap` pattern (https://github.com/AngelList/protip/commit/728962cb1e852bfd746adbc80fd5ff04e1b714a5#diff-12dc79bbbe218018607bed4ecc30c760L31)? `assign_attributes` ends up having a decent bit of overhead.